pixel/image.go

```diff
@@ -99,7 +108,21 @@ func (img Image[T]) Size() (int, int) {
 func (img Image[T]) setPixel(index int, c T) {
 	var zeroColor T
 
-	if zeroColor.BitsPerPixel()%8 == 0 {
+	switch {
+	case zeroColor.BitsPerPixel() == 1:
+		// Monochrome.
+		x := int16(index) % img.width
+		y := int16(index) / img.width
+		offset := x + (y/8)*img.width
+		ptr := (*byte)(unsafe.Add(img.data, offset))
+		rgb := c.RGBA()
+		if rgb.R != 0 {
```

Comment:
Perhaps this is already optimized away by the compiler, but in general a comparison with zero will be most efficient:
```suggestion
		var zeroColor T
		if c != zeroColor {
```

Reply:
Done.